packages/block-library/src/image/edit.native.js

```diff
@@ -385,43 +452,24 @@ export class ImageEdit extends Component {
 
 	setMappedAttributes( { url: href, ...restAttributes } ) {
 		const { setAttributes } = this.props;
+		let linkDestination;
+		if ( restAttributes.linkDestination ) {
+			linkDestination = restAttributes.linkDestination;
+		} else if ( ! href && ! restAttributes.linkDestination ) {
```

Comment:
Since we are always setting local variable `linkDestination` (thus always overwriting its spread counterpart in `restAttributes`), maybe we can destructure it as a function parameter (along with `href`) instead of introducing the local variable. Wdyt?

Reply:
Won't the second part of this (i.e. `! restAttributes.linkDestination`) always be true here, since we are in an else from the condition previous condition?

Reply:
That makes sense to me. I addressed this in https://github.com/WordPress/gutenberg/pull/34846/commits/0749ad210148a99668994d48a11268aa3d92eecc, which changed the conditional logic to still require `&& ! restAttributes.linkDestination`. Let me know if you'd prefer to see this managed in a different manner.